### PR TITLE
feat: Update the GitHub setup flow to include the deployment detection strategy PUL-1602

### PR DESCRIPTION
**Pull request by claudiacarpinteiro**
Update the GitHub setup flow to include the deployment detection strategy. Choosing the deployment detection strategy during the GitHub integration setup enables historical data loading for semantic versioning, which wasn't previously supported.

### :eyes: Live preview
https://feat-gh-setup-add-deploy-detection--docs-pulse.netlify.app/one-click-integrations/github-integration/

### :construction: To do
-   [x] Update content to reflect the new step during setup
-   [x] Update screenshots
-   [x] Request validation
-   [x] Wait for feature release

### CODE REVIEW

File: docs/one-click-integrations/github-integration.md [hunk 1]
```diff
@@ -14,32 +14,35 @@ Pulse integrates with GitHub Cloud to receive data about changes and deployments
...
+    ![Pulse GitHub integration set up successfully](images/github-ok.png)
+
+    If there is an error please [contact support](mailto:pulsesupport@codacy.com).
 
 ## Automatic deployment detection strategies {: id="deployment-detection-strategy"}
 
 The following is a detailed description of how the Pulse GitHub integration automatically detects deployment using each detection strategy:
 
-### Use merged pull requests (default)
+### Use merged pull requests (based on default branch)
```
Comment (mrfyda) [inline docs/one-click-integrations/github-integration.md:45]: I think the default meant that this is the default strategy 😁

File: docs/one-click-integrations/github-integration.md [hunk 2]
```diff
@@ -69,7 +72,8 @@ The following is a detailed description of how the Pulse GitHub integration auto
 
 -   Pulse associates all GitHub teams of the person who creates a Git tag with the corresponding deployment, excluding teams with less than two members. Pulse only takes changes to GitHub teams into account on Git tags created after those changes.
 
-### Don't detect deployments automatically
+### Use the CLI or API (don't detect deployments automatically)
```
Comment (mrfyda) [inline docs/one-click-integrations/github-integration.md:75]: 💪

File: docs/one-click-integrations/github-integration.md (hunk 1 above)
Comment (claudiacarpinteiro) [reply docs/one-click-integrations/github-integration.md:45]: Hummm... I used the name of the option on the UI, according to the [mockup](https://www.figma.com/file/qWJnZBS0q3YRwrQXMyRwGh/%5BPUL-623%5D-Semantic-Versioning-historical-import?node-id=1%3A171&t=MYpjL5fwdnAXLQdX-0). 
I can remove the "default" part here, if it's confusing. WDYT?

**Comment by mrfyda** [comment]
I see. I didn't remember that
nvm the previous comment then 👍

File: docs/one-click-integrations/github-integration.md (hunk 1 above)
Comment (claudiacarpinteiro) [reply docs/one-click-integrations/github-integration.md]: @mrfyda After all, in the development environment we have "(default)" (which is different from the mockup). I left a comment for @nadiia-balaian [in Figma](https://www.figma.com/file/qWJnZBS0q3YRwrQXMyRwGh?node-id=1842:10086#366964241). 🙏

File: docs/one-click-integrations/github-integration.md [hunk 3]
```diff
@@ -14,32 +14,37 @@ Pulse integrates with GitHub Cloud to receive data about changes and deployments
...
-    !!! important
-        You can only install the Pulse GitHub App on an organization and not on your personal account.
+1.  Click **Install GitHub App** and follow the instructions on the GitHub UI to install the app on your **organization**.
 
     ![Installing the Pulse GitHub App](images/github-installing.png)
 
-1.  Wait until you get a confirmation that Pulse successfully created the integration and the webhook on GitHub.
-
-    ![Pulse GitHub integration set up successfully](images/github-ok.png)
+1.  Confirm that Pulse successfully created the integration and the webhook on GitHub. If there is an error please [contact support](mailto:pulsesupport@codacy.com).
```
Comment (nicklem) [inline docs/one-click-integrations/github-integration.md:29]: A couple suggestions here:
* By "integration" are we referring to the GitHub app here? If so, we can reword to clarify that.
* I also added links.

```suggestion
1.  Confirm that Pulse successfully [installed the app](https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/reviewing-your-organizations-installed-integrations) and created the [webhook](https://docs.github.com/en/developers/webhooks-and-events/webhooks/about-webhooks) on GitHub. If there is an error please [contact support](mailto:pulsesupport@codacy.com).
```

**Review by nicklem — APPROVED** [review]
Left a couple small comments. LGTM otherwise!

File: docs/one-click-integrations/github-integration.md (hunk 3 above)
Comment (claudiacarpinteiro) [reply docs/one-click-integrations/github-integration.md:29]: Thanks, @nicklem! Excellent feedback. 👍

**Pull request merged**